pkg/transparentproxy/istio/istio.go

```diff
@@ -0,0 +1,99 @@
+package istio
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/viper"
+	uninstall "istio.io/istio/tools/istio-clean-iptables/pkg/cmd"
+	install "istio.io/istio/tools/istio-iptables/pkg/cmd"
+	"istio.io/istio/tools/istio-iptables/pkg/constants"
+
+	"github.com/kumahq/kuma/pkg/transparentproxy/config"
+)
+
+type IstioTransparentProxy struct {
+	// output redirection
+	stdout *os.File
+	stderr *os.File
+	reader *os.File
+}
+
+func NewIstioTransparentProxy() *IstioTransparentProxy {
+	return &IstioTransparentProxy{}
+}
+
+func (tp *IstioTransparentProxy) Setup(cfg *config.TransparentProxyConfig) (string, error) {
+	viper.Set(constants.EnvoyPort, cfg.RedirectPortOutBound)
+	viper.Set(constants.InboundCapturePort, cfg.RedirectPortInBound)
+	viper.Set(constants.ProxyUID, cfg.UID)
+	viper.Set(constants.ProxyGID, cfg.GID)
+	viper.Set(constants.InboundInterceptionMode, "REDIRECT")
+	viper.Set(constants.InboundPorts, "*")
+	viper.Set(constants.LocalExcludePorts, cfg.ExcludeInboundPorts)
+	viper.Set(constants.ServiceCidr, "*")
+	viper.Set(constants.LocalOutboundPortsExclude, cfg.ExcludeOutboundPorts)
+	viper.Set(constants.DryRun, cfg.DryRun)
+	viper.Set(constants.SkipRuleApply, false)
+	viper.Set(constants.RunValidation, false)
+
+	tp.redirectStdOutStdErr()
+	defer func() {
+		tp.restoreStdOutStderr()
+	}()
+
+	savedArgs := os.Args[1:]
+	os.Args = os.Args[:1]
+	defer func() {
+		os.Args = append(os.Args, savedArgs...)
+	}()
+
+	if err := install.GetCommand().Execute(); err != nil {
```

Comment:
instead of messing around with global `os.Args` and `os.Stdout`, can we do it like with tests?
```
cmd := install.GetCommand()

cmd.SetArgs(...)

stdout := &bytes.Buffer{}
stderr := &bytes.Buffer{}
cmd.SetOut(stdout)
cmd.SetErr(stderr)

err := cmd.Execute()

stdout.Bytes()
```

Reply:
Unfortunately no, because:
 - viper validates that everything in os.Args is a valid flag, and it does not expect to find there our own flags
  - the execute functions in istio-iptabes are directly writing to os.Stdout, there is no way around this